conan/tools/cmake/cmake_initial_cache.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+from conan.tools.cmake.presets import load_cmake_presets, get_configure_preset
+from conans.util.files import save, save_append
+
+
+def _format_cache_variables(cache_variables):
+    ret = []
+    for name, value in cache_variables.items():
+        if isinstance(value, bool):
+            type_ = "BOOL"
+            v = "ON" if value else "OFF"
+        elif value in ("ON", "OFF"):
+            type_ = "BOOL"
+            v = value
+        else:
+            type_ = "STRING"
+            v = f'"{value}"'
+        ret.append(f'set({name} {v} CACHE {type_} "" FORCE)')
+    return "\n".join(ret)
+
+
+def get_initial_cache_path(conanfile):
+    return os.path.join(conanfile.generators_folder, "CMakeInitialCache.cmake")
+
+
+def save_cmake_initial_cache(conanfile):
+    """
+    Save a *.cmake build helper file to save all the CACHE variables (-DXXX vars) in one
+    file instead of passing all of them via CLI.
+
+    Note: it needs the cmake presets file to load all the existing cache variables
+    already defined.
+
+    :param conanfile: ``ConanFile`` instance
+    """
+    cmake_presets = load_cmake_presets(conanfile.generators_folder)
+    configure_preset = get_configure_preset(cmake_presets, conanfile)
+    toolchain_file = configure_preset.get("toolchainFile")
+    cache_variables = configure_preset["cacheVariables"]
```

Comment:
If we go this way, the initial cache doesn't take the values from the Presets, but instead of? Can the presets also define a cache file?

Reply:
Well, I don't think so. Presets has all the information about the cache variables that's correct, but it couldn't call the initial cache file directly (as far as I know), so they are different ways to load the same cache variables. Then, for the developer flow, you could run only the presets one (it already has all the cache variables) or call the initial cache file at first.